### DISCUSSION TITLE
How to reference another widget?

### DISCUSSION THREAD

**Discussion by mjehrhart** [emilk/egui#1496] [Q&A]
Hey,

I'm not sure what to call this, so sorry for that.  I want to have a button that when clicked will hide/show another widget button. I've been looking through the examples but I think I've confused myself. I feel like this is something basic that I missed somewhere.

Edit:: I added a screenshot to help.  I want to click on the filter icon(top right) and be able to hide/unhide the ui.group in the white box.

<img width="1204" alt="Screen Shot 2022-04-14 at 6 26 02 PM" src="https://user-images.githubusercontent.com/97703291/163503261-876610a6-3944-4a54-807f-f37dd40502d4.png">

Is there a way to do this? If so, could you please write up a small simple test example?

Thanks,
Matthew

**Comment by emilk** [comment]
``` rust
struct MyState {
    visible: bool
}

impl MyState {
    fn ui(&mut self, ui: &mut egui::Ui) {
        ui.checkbox(&mut self.visible, "Is the other stuff visible?");
        ui.add_visible_ui(|ui| {
            ui.label("Only visible some times!):
        });
    }
}
```

**Reply by mjehrhart** [reply]
Thanks.  I had that code at one point - completely forgot about it.  Thanks.
